bin/reprocessing_routines.sh

```diff
@@ -112,9 +112,38 @@ get_magetab_for_experiment() {
 
 # 7: see directory, see its contents, can write (fg_atlas only)
 # 5: see directory, see its contents: public read-only directory
+
+## get privacy status for any experiments
+## -MTAB- experiments loaded by biostudies-AE/Annotare uis checked internal file
+## -GEOD-/-ERAD-/-ENAD- are loaded as public from now on
+get_biostudies_privacy_status() {
+    expAcc=$1
+    privStatusFile=$2
+    exp_import=$(echo $expAcc | awk -F"-" '{print $2}')
+
+    if [ $exp_import == "MTAB" ]; then
+        statusStudy=$(awk -v p="$expAcc" '$1 == p {print $2}' $privStatusFile)
+
+        if [ "$statusStudy" == "true" ]; then
+            privacyStatus="public"
+        elif [ "$statusStudy" == "false" ]; then
+            privacyStatus="private"
+        else 
+            >&2 echo "get_biostudies_privacy_status could not determine privacy status for $1, received: $statusStudy"
+            exit 1
+        fi
+    else
+        # if not MTAB, ie. GEOD or ENAD or ERAD are all loaded as public
+        privacyStatus="public"
+    fi
+
+    echo $privacyStatus
+}
+
 experiment_directory_permissions_from_peach_api_privacy() {
```

Comment:
Perhaps a name change to remove the reference to the peach API? In this repo, the function is only used here in this script.  The same function is renamed `experiment_directory_permissions_from_biostudies_privacy` in the atlas-prod repo in [this PR](https://github.com/ebi-gene-expression-group/atlas-prod/pull/254).